app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java

```diff
@@ -72,4 +79,48 @@ public int hashCode() {
         return this.name.hashCode() + this.publicKey.hashCode() + this.privateKey.hashCode();
     }
 
+    /**
+     * Serialize the created proxy instead of this instance.
+     */
+    private Object writeReplace() {
+        return new KeyPair.SerializationProxy(this);
```

Comment:
I don't know actually I added them by example of other serializable objects that were already made